modules/fltPage/components/PriceInfoCard/PriceInfoCardContainer.tsx

```diff
@@ -0,0 +1,97 @@
+import { FunctionComponent, useEffect, useState } from "react";
+import PriceInfoCardHeader from "./PriceInfoCardHeader";
+import PriceInfoCardMetric from "./PriceInfoCardMetric";
+import FLTChart from "./Chart/FLTChart";
+import { useLeveragedTokenNAV } from "../../../../components/v1/swr/useLeveragedTokenNAV";
+import { getProvider } from "../../../../components/v1/Wallet";
+import { chain } from "wagmi";
+import { ethers } from "ethers";
+import { LeveragedTokenTimeframeData, useLeveragedTokenHistoricalData } from "../../../../components/v1/swr/useLeveragedTokenHistoricalData";
+import ChartFilters from "./Chart/ChartFilters";
+import { Timeframe } from "../../../../components/v1/swr/snapshot";
+
+type PriceInfoCardContainerProps = {};
+
+const PriceInfoCardContainer: FunctionComponent<PriceInfoCardContainerProps> = ({}) => {
+    const [activeTimeframe, setActiveTimeframe] = useState<Timeframe>(Timeframe.TwoWeekly);
+
+    // Get data
+    const data = useLeveragedTokenHistoricalData(chain.arbitrumOne.id, "0x46D06cf8052eA6FdbF71736AF33eD23686eA1452");
+
+    // Set selected data state
+    const [selectedData, setSelectedData] = useState<LeveragedTokenTimeframeData>();
+
+    // Set default initial data
+    if (!selectedData && data.twoWeekly) {
+        setSelectedData(data.twoWeekly);
+    }
+
+    // Read onchain data (temporary until real data is provided)
+    const navResponse = useLeveragedTokenNAV({ token: "0x46D06cf8052eA6FdbF71736AF33eD23686eA1452", vault: "0xf7EDB240DbF7BBED7D321776AFe87D1FBcFD0A94", provider: getProvider({ chainId: chain.arbitrumOne.id }) });
+    const latestNav = parseFloat(ethers.utils.formatUnits(navResponse.data ? navResponse.data : 0, 6));
+    const latestNavChange = selectedData ? latestNav - selectedData.oldestNAV : 0;
+    const latestPercentChange = selectedData ? ((latestNav - selectedData.oldestNAV) / selectedData.oldestNAV) * 100 : 0;
+
+    // Set tooltip selected metric data
+    const [currentNav, setCurrentNav] = useState(0);
+    const [currentNavChange, setCurrentNavChange] = useState(0);
+    const [currentNavPercentChange, setCurrentNavPercentChange] = useState(0);
+    const [currentDate, setCurrentDate] = useState("");
+
+    // Set default initial nav data
+    if (currentNav === 0 && currentNavChange === 0 && latestNav != 0 && latestNavChange != 0 && latestPercentChange != 0) {
+        setCurrentNav(latestNav);
+        setCurrentNavChange(latestNavChange);
+        setCurrentNavPercentChange(latestPercentChange);
+    }
+
+    const resetCurrentNav = () => {
+        setCurrentNav(latestNav);
+        setCurrentNavChange(latestNavChange);
+        setCurrentNavPercentChange(latestPercentChange);
+        setCurrentDate("");
+    };
+
+    // Listen to timeframe change (this is temporary & subject to changes until real data is ready)
+    useEffect(() => {
```

Comment:
please insert your data to dependencies list

then convert your code to something like this

```
useEffect(()=>{
if (data){
switch (activeTimeFrame){
            case Timeframe.Daily: setSelectedData(data.daily) break;
....

}
}
},[])